src/main/java/stsjorbsmod/cards/wanderer/Determination.java

```diff
@@ -37,7 +38,17 @@ public Determination() {
     public void use(AbstractPlayer p, AbstractMonster m) {
         addToBot(new RememberSpecificMemoryAction(p, PrideMemory.STATIC.ID));
         if (!upgraded) {
-            addToBot(new ApplyPowerAction(p, p, new FragilePower(p, magicNumber), magicNumber));
+            if (p.hasPower(FragilePower.POWER_ID)) {
+                FragilePower fragilePower = (FragilePower) p.getPower(FragilePower.POWER_ID);
+                fragilePower.flash();
+                fragilePower.amount = 1;
+                fragilePower.updateDescription();
+            } else if (p.hasPower(SnappedPower.POWER_ID)) {
+                p.getPower(SnappedPower.POWER_ID).flash();
+            } else if (!p.hasPower(FragilePower.POWER_ID) && !p.hasPower(SnappedPower.POWER_ID)) {
+                // apply FragilePower only if the player doesn't already have FragilePower or SnappedPower.
```

Comment:
This comment isn't adding anything; what it's saying is already clear from the code